src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java

```diff
@@ -154,7 +162,14 @@ public static Type getTypeFromJson(@Nonnull JsonNode jsonNode) {
           field.isObject(), "Cannot parse struct field from non-object: %s", field);
 
       int id = JsonUtil.getInt(ID, field);
-      String name = JsonUtil.getString(NAME, field);
+
+      /* TypeToMessageType throws on empty field name, use a backslash to represent it and escape remaining backslash. */
```

Comment:
iceberg spec allows empty field name?

Reply:
Iceberg spec doesn't explicitly disallow it. Snowflake Iceberg DML supports it.

Reply:
got it, do have an IT test with empty names then, reactivating to make sure you see this!

no need to block this PR for empty field name IT test coverage.

Reply:
In the IT of this PR. [ref](https://github.com/snowflakedb/snowflake-ingest-java/blob/af46e607320cbb2f8f79402d04d67d2d72c94a90/src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java#L180).